### PR TITLE
Adding all arguments to Key() constructor.

### PR DESCRIPTION
**Pull request by dhermes**
~~**NOTE**: This has #14 as diffbase.~~

Added documentation for each argument to the docstring. For most arguments, just raising `NotImplementedError` for now, but the `app` and `namespace` inputs are supported. The current handling of `app` likely needs tweaking so I added a note to the README about how I think we might handle it.

### CODE REVIEW

File: src/google/cloud/ndb/key.py [hunk 1]
```diff
@@ -12,12 +12,196 @@
...
+    * ``key.app()``: the Google Cloud Platform project (formerly called the
+      application ID)
+    * ``key.id()``: the string or integer ID in the last ``(kind, id)`` pair,
+      or :data:`None` if the key is incomplete
+    * ``key.string_id()``: the string ID in the last ``(kind, id)`` pair,
+      or :data:`None` if the key has an integer ID or is incomplete
+    * ``key.integer_id()``: the integer ID in the last ``(kind, id)`` pair,
+      or :data:`None` if the key has a string ID or is incomplete
+    * ``key.namespace()``: the namespace
+    * ``key.kind()``: a shortcut for ``key.pairs()[-1][0]``
```
Comment (crwilcox) [inline src/google/cloud/ndb/key.py]: It seems odd to describe this as `__ in the last (kind, id) pair` and also as `a shortcut for key.pairs()[-1][0]`.  I prefer `kind in the last (kind, id) pair` for the docstr I think, even though they should be equivalent?

File: src/google/cloud/ndb/key.py [hunk 2]
```diff
@@ -12,12 +12,196 @@
...
+    * ``key.parent()``: a key constructed from all but the last ``(kind, id)``
```
Comment (crwilcox) [inline src/google/cloud/ndb/key.py]: This is a bit confusing. Is the parent the first `(kind, id)` pair? Or is it all but the last? Previous comments made me believe it was the former.

File: src/google/cloud/ndb/key.py (hunk 1 above)
Comment (dhermes) [reply src/google/cloud/ndb/key.py]: SGTM. This was copy-pasted and I also didn't like the use of the index `-1`

File: src/google/cloud/ndb/key.py (hunk 2 above)
Comment (dhermes) [reply src/google/cloud/ndb/key.py]: I'll try to clean it up and make it less confusing. The parent of

```
(kind1, id1), (kind2, id2), (kind3, id3)
```

is

```
(kind1, id1), (kind2, id2)
```

i.e. `path[:-1]`.

File: src/google/cloud/ndb/key.py [hunk 3]
```diff
@@ -108,9 +108,12 @@ class Key:
     * ``key.integer_id()``: the integer ID in the last ``(kind, id)`` pair,
       or :data:`None` if the key has a string ID or is incomplete
     * ``key.namespace()``: the namespace
-    * ``key.kind()``: a shortcut for ``key.pairs()[-1][0]``
+    * ``key.kind()``: The "kind" of the key,from the last of the
```
[the remark this answers is not in the document]
Comment (dhermes) [reply src/google/cloud/ndb/key.py]: Absolutely, thanks

**Pull request merged**